Sources/FluentPostgreSQL/PostgreSQLDriver.swift

```diff
@@ -56,23 +58,7 @@ public class PostgreSQLDriver: Fluent.Driver {
 
         switch query.action {
         case .create:
-            // fetch the last inserted value
-            do {
-                let lastval = try _execute("SELECT LASTVAL();", [], connection)
-                // check if it contains an id
-                if let id = lastval[0, "lastval"]?.int {
-                    // return the id to fluent to
-                    // be the model's new id
-                    return .number(.int(id))
-                }
-            }
-            catch PostgreSQL.DatabaseError.invalidSQL(message: _) {
-                // When receiving a invalidSQL error, there is basically no LASTVAL, so ignore.
-            }
-
-            // no id found, return whatever
-            // the results are
-            return result
+            return (result[idKey]?.array as? [Node])?.first ?? result
```

Comment:
As this is a Fluent 1.x driver, I think it's safe to assume that at most one item will be inserted into the database. The create action seems to be geared this anyway.